src/app/public/modules/colorpicker/colorpicker.component.ts

```diff
@@ -310,9 +378,71 @@ export class SkyColorpickerComponent implements OnInit, OnDestroy {
     }
   }
 
-  private closeDropdown() {
-    this.dropdownController.next({
-      type: SkyDropdownMessageType.Close
+  private createAffixer(): void {
+    const affixer = this.affixService.createAffixer(this.colorpickerRef);
+
+    affixer.placementChange
+      .takeUntil(this.pickerUnsubscribe)
+      .subscribe((change) => {
+        this.isVisible = (change.placement !== null);
+      });
+
+    affixer.affixTo(this.triggerButtonRef.nativeElement, {
+      autoFitContext: SkyAffixAutoFitContext.Viewport,
+      enableAutoFit: true,
+      horizontalAlignment: 'left',
+      isSticky: true,
+      placement: 'below'
+    });
+
+    this.affixer = affixer;
+  }
+
+  private destroyAffixer(): void {
+    /*istanbul ignore else*/
+    if (this.affixer) {
+      this.affixer.destroy();
+      this.affixer = undefined;
+    }
+  }
+
+  private createOverlay(): void {
+    const overlay = this.overlayService.create({
+      enableClose: false,
+      enablePointerEvents: false,
```

Comment:
The current colorpicker demo lets you click the document, but this new overlay blocks those clicks. Should `enablePointerEvents` be `true`? Or is that changing?

Reply:
Based on [this issue](https://github.com/blackbaud/skyux-colorpicker/issues/14), we want to avoid multiple colorpickers from being opened at once; preventing pointer events solves that problem and makes the colorpicker behave more like a modal, which is better for accessibility. Open to your opinion, however.

Reply:
Yeah, that's reasonable. Thanks for the explanation.